src/wallet/rpcwallet.cpp

```diff
@@ -3155,38 +3155,20 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
 
     std::string defaultType = ADDR_TYPE_SAPLING;
 
-    if (fHelp || params.size() > 1)
+    if (fHelp || params.size() > 0)
         throw runtime_error(
-            "z_getnewaddress ( type )\n"
+            "z_getnewaddress\n"
             "\nReturns a new shielded address for receiving payments.\n"
-            "\nWith no arguments, returns a Sapling address.\n"
-            "\nArguments:\n"
-            "1. \"type\"         (string, optional, default=\"" + defaultType + "\") The type of address. One of [\""
-            + ADDR_TYPE_SPROUT + "\", \"" + ADDR_TYPE_SAPLING + "\"].\n"
             "\nResult:\n"
             "\"zcashaddress\"    (string) The new shielded address.\n"
```

Comment:
should zcashaddress here be arrowaddress or something, while you're in here tidying?

Reply:
good call - i'll update